### DISCUSSION TITLE
Deleting old version of log4j after upgrade

### DISCUSSION THREAD

**Discussion by rjkroll** [nextgenhealthcare/connect#5626] [Q&A] [answered]
Hi,

We recently upgraded to the the latest version of Mirth to resolve the older CVE's regarding the log4j exploit from last year.

The upgrade went fine, but our security scans are still complaining that they see v.1.x of log4j still on the server in the following paths.  Is it safe to delete these jars/files without affecting Mirth or the Administrator?

C:\Program   Files\Mirth Connect\manager-lib\log4j-1.2.16.jar 
C:\Program   Files\Mirth Connect\client-lib\log4j-1.2.16.jar 
C:\Program   Files\Mirth Connect\server-lib\log4j-1.2.16.jar 
C:\Program   Files\Mirth Connect Administrator Launcher\cache\3.12.0\core\log4j-1.2.16.jar

Thank you in advance

**Comment by pacmano1** [comment, marked as answer]
Files in cache you can delete.

You can help yourself by doing a clean install on a new server and seeing if the old log4j files get installed.  If they don't, you can remove them.

For example in Linux:

```shell
root@primary:/usr/local/mirthconnect# find  . -name "log4j*" -print
```
returns:
```
./server-lib/log4j
./server-lib/log4j/log4j-api-2.17.2.jar
./server-lib/log4j/log4j-1.2-api-2.17.2.jar
./server-lib/log4j/log4j-core-2.17.2.jar
./server-lib/donkey/log4j-api-2.17.2.jar
./server-lib/donkey/log4j-1.2-api-2.17.2.jar
./server-lib/donkey/log4j-core-2.17.2.jar
./conf/log4j2.properties
./client-lib/log4j-api-2.17.2.jar
./client-lib/log4j-1.2-api-2.17.2.jar
./client-lib/log4j-core-2.17.2.jar
```

**Reply by rjkroll** [reply]
Thanks.   I was able to find and delete the ones in the cache directory, but I don't even see the JARs from the scan they performed in the Mirth Connect lib directories, I only see the new versions of the log4j jar, so I guess I will have to ask them how their scan is finding those files when I can't see them in the folders or search the whole drive to find them.

**Reply by pacmano1** [reply]
Sorta funny actually.   Way to go security team.
```
I don't even see the JARs from the scan they performed in the Mirth Connect lib directories
```
